ext/node/polyfills/01_require.js

```diff
@@ -994,25 +1002,28 @@ Module._extensions[".js"] = function (module, filename) {
   if (StringPrototypeEndsWith(filename, ".js")) {
     const pkg = ops.op_require_read_closest_package_json(filename);
     if (pkg && pkg.exists && pkg.typ === "module") {
-      throw new Error(
-        requireEsmErrorText(filename, moduleParentCache.get(module)?.filename),
+      throw createRequireEsmError(
+        filename,
+        moduleParentCache.get(module)?.filename,
       );
     }
   }
 
   module._compile(content, filename);
 };
 
-function requireEsmErrorText(filename, parent) {
-  let message = `[ERR_REQUIRE_ESM]: require() of ES Module ${filename}`;
+function createRequireEsmError(filename, parent) {
+  let message = `require() of ES Module ${filename}`;
 
   if (parent) {
     message += ` from ${parent}`;
   }
 
   message +=
     ` not supported. Instead change the require to a dynamic import() which is available in all CommonJS modules.`;
-  return message;
+  const err = new Error(message);
+  err.code = "ERR_REQUIRE_ESM";
```

Comment:
Fixes https://github.com/denoland/deno/issues/16913 properly this time.

That said, in the future we should go through and rewrite all these errors to be more similar to what Node does.